testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/policies/ClientPoliciesConditionTest.java

```diff
@@ -382,6 +384,52 @@ public void testClientScopesCondition() throws Exception {
         }
     }
 
+    @Test
+    public void testClientAttributesCondition() throws Exception {
+        // register profiles
+        String json = (new ClientProfilesBuilder()).addProfile(
+                (new ClientProfileBuilder()).createProfile(PROFILE_NAME, "Het Eerste Profiel")
+                        .addExecutor(PKCEEnforcerExecutorFactory.PROVIDER_ID,
+                                createPKCEEnforceExecutorConfig(Boolean.TRUE))
+                        .toRepresentation()
+        ).toString();
+        updateProfiles(json);
+
+        // register policies
+        json = (new ClientPoliciesBuilder()).addPolicy(
+                (new ClientPolicyBuilder()).createPolicy(POLICY_NAME, "Het Eerste Beleid", Boolean.TRUE)
+                        .addCondition(ClientAttributesConditionFactory.PROVIDER_ID,
+                                createClientAttributesConditionConfig(new HashMap<String, String>() {
+                                    {
+                                        put("attr1", "Apple");
+                                        put("attr2", "Orange");
+                                    }
+                                }))
+                        .addProfile(PROFILE_NAME)
+                        .toRepresentation()
+        ).toString();
+        updatePolicies(json);
+
+        String clientAlphaId = generateSuffixedName("Alpha-App");
+        String clientSecret = "secret";
+        createClientByAdmin(clientAlphaId, (ClientRepresentation clientRep) -> {
+            clientRep.setSecret(clientSecret);
+            clientRep.setAttributes(new HashMap<String, String>() {
+                {
+                    put("attr1", "Apple");
+                    put("attr2", "Orange");
+                }
+            });
+        });
+
+        try {
+            failLoginByNotFollowingPKCE(clientAlphaId);
+            successfulLoginAndLogoutWithPKCE(clientAlphaId, clientSecret, TEST_USER_NAME, TEST_USER_PASSWORD);
```

Comment:
It might be better to add a test where this conditon's evaluation resuit is false.

Reply:
Added the test.